docker/install/ubuntu_install_caffe.sh

```diff
@@ -22,13 +22,40 @@ set -o pipefail
 
 apt-get update --fix-missing
 
-# The precompiled caffe dependents on tzdata.
-# While installing tzdata in docker, we need set the time zone manually,
-# which will cause the container to hang during installation.
-# So in order to avoid manually selecting the time zone, set as following:
-export DEBIAN_FRONTEND=noninteractive
-apt-get install -y tzdata
+# Install dependencies
+apt-get install -y --no-install-recommends libboost-filesystem-dev libboost-python-dev \
+    libboost-system-dev libboost-thread-dev libboost-regex-dev protobuf-compiler \
+    libprotobuf-dev libhdf5-serial-dev libopenblas-dev libgflags-dev libgoogle-glog-dev
+rm -rf /var/lib/apt/lists/*
```

Comment:
Why needs this command?

Reply:
Those packages are the minimum requirement for building caffe from source.
So I put them.

Reply:
the `rm -rf /var/lib/apt/lists/*` part seems a little weird. can you elaborate/link to these instructions?

Reply:
That's just a best practice for Dockerfiles. 
[Best practices for writing Dockerfiles | Docker Documentation](https://docs.docker.com/develop/develop-images/dockerfile_best-practices/#run)

It might be unnecessary for this script. I'll remove it.

Reply:
ah, didn't know this but it definitely makes sense. it wasn't the convention we were using before, but we should adopt it. i've filed #9825 to track this.